src/ol/layer/BaseVector.js

```diff
@@ -214,6 +215,17 @@ class BaseVectorLayer extends Layer {
     return this.updateWhileInteracting_;
   }
 
+  /**
+   * Render declutter items for this layer
+   * @param {import("../PluggableMap.js").FrameState} frameState Frame state.
+   */
+  renderDeclutter(frameState) {
+    if (!frameState.declutterTree) {
+      frameState.declutterTree = new RBush(9);
+    }
```

Comment:
`Renderer#renderDeclutter` will always be called by `Layer#renderDeclutter`. The reason why I put it in the layer instead of the renderer is because decluttering is not renderer specific, i.e. it could also be implemented by a custom renderer.